sphinx/environment/collectors/asset.py

```diff
@@ -79,8 +79,11 @@ def process_doc(self, app: Sphinx, doctree: nodes.document) -> None:
 
                 # Update `node['uri']` to a relative path from srcdir
                 # from a relative path from current document.
+                original_uri = node['uri']
                 node['uri'], _ = app.env.relfn2path(imguri, docname)
                 candidates['*'] = node['uri']
+                if node['uri'] != original_uri:
+                    node['original_uri'] = original_uri
```

Comment:
@alvinhochun -- is this needed for this PR? If not, could it be in a different PR please?

A

Reply:
Hi, I don't exactly remember the context for this commit, if that change make sense to you would you mind splitting that commit into its own PR for me? Otherwise you can try to revert or remove that commit from this PR.

(FWIW it *might* have been related to image URIs inside substitutions, but I'm just guessing from the test cases. See if the tests pass without this commit.)

Reply:
The context might be [this comment](https://github.com/sphinx-doc/sphinx/pull/9454#issuecomment-884172757).